### PR TITLE
Add Missing Python Version of TickDataFilteringAlgorithm (#7219)

### LINKED ISSUE #7219: Missing Python Version of TickDataFilteringAlgorithm
#### Expected Behavior

We have TickDataFilteringAlgorithm.cs and TickDataFilteringAlgorithm.py 

#### Actual Behavior

We have TickDataFilteringAlgorithm.cs but no TickDataFilteringAlgorithm.py

#### Potential Solution

Add TickDataFilteringAlgorithm.py

#### Reproducing the Problem

N/A

#### System Information

N/A

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [ ] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by hungnguyen9**
#### Description
Missing Python Version of TickDataFilteringAlgorithm #7219 

#### Related Issue
Close #7219 

#### Requires Documentation Change

No

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

### CODE REVIEW

**Review by AlexCatarino — CHANGES_REQUESTED** [review]
Thank you for the contribution, @hungnguyen9 .

I ran this algorithm on QuantConnect Cloud and it raised the following exception:

```
During the algorithm initialization, the following exception has occurred: object does not implement ISecurityDataFilter
  at Initialize
```

**Comment by hungnguyen9** [comment]
> Thank you for the contribution, @hungnguyen9 .
> 
> I ran this algorithm on QuantConnect Cloud and it raised the following exception:
> 
> ```
> During the algorithm initialization, the following exception has occurred: object does not implement ISecurityDataFilter
>   at Initialize
> ```

**Comment by hungnguyen9** [comment]
> Thank you for the contribution, @hungnguyen9 .
> 
> I ran this algorithm on QuantConnect Cloud and it raised the following exception:
> 
> ```
> During the algorithm initialization, the following exception has occurred: object does not implement ISecurityDataFilter
>   at Initialize
> ```

Dear @AlexCatarino,

Thank you very much for your comment. I have modified the code. Could you please check again?

File: Algorithm.Python/TickDataFilteringAlgorithm.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# <summary>
+# Exchange filter class
+# </summary>
+class ExchangeDataFilter(SecurityDataFilter):
+
+    # <summary>
+    # Save instance of the algorithm namespace
+    # </summary>
+    # <param name="algo"></param>
+    def ExchangeDataFilter(self, algo):
```
Comment (AlexCatarino) [inline Algorithm.Python/TickDataFilteringAlgorithm.py:58]: This should be `def __init__` and call `super()` this the class inherits from `SecurityDataFilter`. However, this class might not need to inherit from `SecurityDataFilter`.

**Review by AlexCatarino — DISMISSED** [review]
Hi @hungnguyen9 ,

Please copy your algorithm, and run a backtest in QuantConnect Cloud. It should not raise exceptions. I ran the latest version and got the following:

```
During the algorithm initialization, the following exception has occurred: Trying to dynamically access a method that does not exist throws a TypeError exception. To prevent the exception, ensure each parameter type matches those required by the 'main.TickDataFilteringAlgorithm'>) method. Please checkout the API documentation.
  at Initialize
    self.Securities["SPY"].DataFilter = ExchangeDataFilter(self)
```

I recommend looking #7201 , since it has added the features that make this algorithm possible.

We have also added Documentation on:
https://www.quantconnect.com/docs/v2/writing-algorithms/securities/filtering-data

**Comment by hungnguyen9** [comment]
> Hi @hungnguyen9 ,
> 
> Please copy your algorithm, and run a backtest in QuantConnect Cloud. It should not raise exceptions. I ran the latest version and got the following:
> 
> ```
> During the algorithm initialization, the following exception has occurred: Trying to dynamically access a method that does not exist throws a TypeError exception. To prevent the exception, ensure each parameter type matches those required by the 'main.TickDataFilteringAlgorithm'>) method. Please checkout the API documentation.
>   at Initialize
>     self.Securities["SPY"].DataFilter = ExchangeDataFilter(self)
> ```
> 
> I recommend looking #7201 , since it has added the features that make this algorithm possible.
> 
> We have also added Documentation on: https://www.quantconnect.com/docs/v2/writing-algorithms/securities/filtering-data

Hi @AlexCatarino,

Thank you for your comment. I have modified my code, ran a backtest in QC cloud and it looked okay. Could you please check my PR again?

Regards,
Hung

**Review by Martin-Molinero — APPROVED** [review]
Thanks @hungnguyen9!

**Pull request merged**